### PR TITLE
Feat/get js text link props

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

contributes to #418 

## What is the new behavior?

Adds JS API & tests

## Other information

Updates generated icon css properties that were missed after an earlier change

### CODE REVIEW

**Comment by Rykus0** [comment]
Sandboxes:
[CSS-in-JS (SC)](https://codesandbox.io/s/basic-text-link-css-in-js-sc-j34zvf)
[MUI](https://codesandbox.io/s/basic-text-link-mui-dv200j?file=/src/App.tsx)

File: packages/headless-styles/src/components/TextLink/textLinkJS.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const defaultOptions = getDefaultTextLinkOptions(options)
+  const props = createTextLinkProps(defaultOptions.href, defaultOptions.tech)
+  const jsStyles = {
+    ...styles.textLinkBase,
+  }
+
+  return {
+    ...props,
+    link: {
+      a11yProps: props.link,
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/TextLink/textLinkJS.ts:16]: Isn't `props.link` a String? If so, we need to wrap this in an Object with the correct attr for the prop name in order for it to work properly.

Comment (caseybaggz) [reply packages/headless-styles/src/components/TextLink/textLinkJS.ts:16]: Something like...
```
a11yProps: {
  href: props.link
}
```

Comment (Rykus0) [reply packages/headless-styles/src/components/TextLink/textLinkJS.ts:16]: `props.link` is the object that contains `cssProps` and `styles`, so we need to assign the rest of the props to their own property.

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**